homeassistant/components/smlight/update.py

```diff
@@ -80,42 +100,46 @@ def __init__(
         self,
         coordinator: SmFirmwareUpdateCoordinator,
         description: SmUpdateEntityDescription,
+        idx: int = 0,
     ) -> None:
         """Initialize the entity."""
         super().__init__(coordinator)
 
         self.entity_description = description
-        self._attr_unique_id = f"{coordinator.unique_id}-{description.key}"
+        device = description.key + (f"_{idx}" if idx else "")
+        self._attr_unique_id = f"{coordinator.unique_id}-{device}"
 
         self._finished_event = asyncio.Event()
         self._firmware: Firmware | None = None
         self._unload: list[Callable] = []
+        self.idx = idx
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self.async_on_remove(self.coordinator.async_add_listener(self._update_listener))
+        self._update_listener()
+
+    @callback
+    def _update_listener(self) -> None:
```

Comment:
`BaseCoordinatorEntity`, which `SmEntity` derives from, provides `_handle_coordinator_update` for this purpose and is registered as a listener in the base implementation of `async_added_to_hass`.

```suggestion
    def _handle_coordinator_update(self) -> None:
```

Reply:
yes, now you mention it, I recall that function ;)